### DISCUSSION TITLE
Spherical Harmonics vs Normal Irrep

### DISCUSSION THREAD

**Discussion by fses91** [e3nn/e3nn#391] [Q&A] [answered]
Hi, 

I have a question about the use of Spherical Harmonics and Irreps in the Convolution example (https://docs.e3nn.org/en/latest/guide/convolution.html).

In this example you are using irreps_input = o3.Irreps("10x0e + 10x1e") (which represent 10 even scalar quantities and 10 even vector quantities), and for the edge_vec you are using o3.spherical_harmonics.

And then you take the tensor product to create some new vector quantities.

Could I also represent the edge_vec just as o3.Irreps("1x1o") which is also a vector (the direction vector between two points) if I am right, because in the documentation it is stated that irreps means "transforms like a vector". Why are you using for some vectors o3.Irreps and for other vectors the o3.spherical_harmonics, and why are you using here lmax=2 for the spherical harmonics, I thought l=1 is enough to represent the vector.

So when do I choose what, or do I have a general misunderstanding?

My background is computer science and not physics and just started with this topic.

Thanks.

**Comment by mariogeiger** [comment, marked as answer]
You can use only the vector of course. Feel free

Spherical harmonics is just a polynomial of x, it gives you a richer representation for the edge. lmax=2 in this context means that as edge attribute you use x **and** x^2 (only the non redundant part of x^2 which is an irrep with l=2)

Edit: I can try to guess the source of the confusion: the spherical harmonics are also used as the Fourier basis for the functions on the sphere. Therefore one might think that SH imply that "you go in the Fourier space". But in our usage here we just use the SH as a polynomial of it's input.

**Reply by fses91** [reply]
Thank you for the fast answer Mario.

Ok I think now I get an intuition, so e.g. you have the features like _length_ and _width_ in meters of a house and create the feature _square meters_, or like a polynomial kernel in SVM.

I hadn't considered this for a direction vector before.

**Comment by blondegeek** [comment]
You can represent the vector as just a 1o object. The advantage of
including higher spherical harmonic projections comes from the fact that
you are summing the contributions from multiple neighbors — e.g. imagine
that you had two convolution neighbors that were in the -r and +r
directions with identical node features. These contributions would sum to
zero for L=1, but not for L=2. Similar arguments can be made for higher Ls.
The higher Ls are sensitive to higher “angular frequencies” — its the same
thing as Nyquist sampling but on the sphere. Hope that helps!

**Reply by fses91** [reply]
Thank you for providing the example, that helped me a lot.
